src/rf/apps/workers/emr.py

```diff
@@ -103,6 +103,35 @@ def create_cluster(layer):
         Steps=get_steps(layer, status_queue),
     )
     log.info(response)
+    return response
+
+
+def check_status(cluster_id):
+    alive_instance_statuses = [
+        'PROVISIONING',
+        'BOOTSTRAPPING',
+        'RUNNING',
+        'RESIZING',
+    ]
+    alive_step_statuses = [
+        'PENDING',
+        'RUNNING',
+        'COMPLETED',
+    ]
+    client = boto3.client('emr')
+    response = client.list_instance_groups(ClusterId=cluster_id)
+    statuses = (instance['Status']['State']
+                for instance in response['InstanceGroups'])
+    instances_alive = all(map(lambda s: True if s in alive_instance_statuses
+                          else False, statuses))
+    if instances_alive:
+        response = client.list_steps(ClusterId=cluster_id)
+        statuses = (step['Status']['State']
+                    for step in response['Steps'])
+        return all(map(lambda s: True if s in alive_step_statuses
+                   else False, statuses))
+    else:
+        return False
```

Comment:
May be simpler just to make 1 API call to http://boto3.readthedocs.org/en/latest/reference/services/emr.html?highlight=emr#EMR.Client.describe_cluster.